src/lib/workspace/AllAnalyses.tsx

```diff
@@ -0,0 +1,338 @@
+import { orderBy } from 'lodash';
+import Path from 'path';
+import React, { useCallback, useMemo, useState } from 'react';
+import { useRouteMatch } from 'react-router';
+import { Link, useHistory } from 'react-router-dom';
+import {
+  Analysis,
+  AnalysisClient,
+  SubsettingClient,
+  useAnalysisList,
+  usePinnedAnalyses,
+} from '../core';
+import { workspaceTheme } from '../core/components/workspaceTheme';
+
+const useStyles = makeStyles({
+  root: {
+    '& .ActionToolbar-ItemList': {
+      width: '100%',
+    },
+    '& .ActionToolbar-Item': {
+      display: 'flex',
+      alignItems: 'center',
+      '&:last-of-type': {
+        marginLeft: 'auto',
+      },
+    },
+    '& .MesaComponent td': {
+      verticalAlign: 'middle',
+    },
+  },
+});
+
+interface Props {
+  analysisClient: AnalysisClient;
+  subsettingClient: SubsettingClient;
+}
+
+export function AllAnalyses(props: Props) {
+  const { analysisClient } = props;
+  const { url } = useRouteMatch();
+  const [selectedAnalyses, setSelectedAnalyses] = useState<Set<string>>(
+    new Set()
+  );
+  const classes = useStyles();
+
+  const {
+    pinnedAnalyses,
+    isPinnedAnalysis,
+    addPinnedAnalysis,
+    removePinnedAnalysis,
+  } = usePinnedAnalyses(analysisClient);
+
+  const { analyses, deleteAnalyses, loading, error } = useAnalysisList(
+    analysisClient
+  );
+
+  const [sortPinned, setSortPinned] = useSessionBackedState<boolean>(
+    true,
+    'eda::allAnalysesPinned',
+    JSON.stringify,
+    JSON.parse
+  );
+
+  const [tableSort, setTableSort] = useSessionBackedState<
+    [string, 'asc' | 'desc']
+  >(['modified', 'desc'], 'eda::allAnalysesSort', JSON.stringify, JSON.parse);
+
+  // const [sortPinned, setSortPinned] = useState(true);
+
+  const datasets = useWdkService(
+    (wdkService) =>
+      wdkService.getAnswerJson(
+        {
+          searchName: 'Studies',
+          searchConfig: {
+            parameters: {},
+          },
+        },
+        {
+          attributes: ['dataset_id'],
+        }
+      ),
+    []
+  );
+  const history = useHistory();
+
+  const removeUnpinned = useCallback(() => {
+    if (analyses == null) return;
+    const idsToRemove = analyses
+      .map((analysis) => analysis.id)
+      .filter((id) => !isPinnedAnalysis(id));
+    deleteAnalyses(idsToRemove);
+  }, [analyses, deleteAnalyses, isPinnedAnalysis]);
+
+  const tableState = useMemo(
+    () => ({
+      rows: sortPinned
+        ? orderBy(
+            analyses,
+            [
+              (analysis) => (isPinnedAnalysis(analysis.id) ? 0 : 1),
+              (analysis) => {
+                const columnKey = tableSort[0];
+                switch (columnKey) {
+                  case 'study':
+                    return (
+                      datasets?.records.find(
+                        (d) => d.id[0].value === analysis.studyId
+                      )?.displayName ?? 'Unknown study'
```

Comment:
There's a second or so's delay when I reload this page, and "Unknown study" is a bit disconcerting.

Could this work?
```suggestion
                      )?.displayName ?? 'Loading...'
```

Reply:
Actually this is the wrong place to do this (see below)